pennylane_lightning/src/StateVector.hpp

```diff
@@ -15,19 +15,764 @@
+     */
+    void applyOperation(const string &opName, const vector<size_t> &wires,
+                        bool inverse = false, const vector<fp_t> &params = {}) {
+        const auto gate = gates_.at(opName);
+        if (gate_wires_.at(opName) != wires.size())
+            throw std::invalid_argument(
+                string("The gate of type ") + opName + " requires " +
+                std::to_string(gate_wires_.at(opName)) + " wires, but " +
+                std::to_string(wires.size()) + " were supplied");
+
+        const vector<size_t> internalIndices = generateBitPatterns(wires);
+        const vector<size_t> externalWires = getIndicesAfterExclusion(wires);
+        const vector<size_t> externalIndices =
+            generateBitPatterns(externalWires);
+
+        gate(internalIndices, externalIndices, inverse, params);
+    }
+
+    /**
+     * @brief Apply multiple gates to the state-vector.
+     *
+     * @param ops Vector of gate names to be applied in order.
+     * @param wires Vector of wires on which to apply index-matched gate name.
+     * @param inverse Indicates whether gate at matched index is to be inverted.
+     * @param params Optional parameter data for index matched gates.
+     */
+    void applyOperations(const vector<string> &ops,
+                         const vector<vector<size_t>> &wires,
+                         const vector<bool> &inverse,
+                         const vector<vector<fp_t>> &params) {
+        const size_t numOperations = ops.size();
+        if (numOperations != wires.size() || numOperations != params.size())
+            throw std::invalid_argument(
+                "Invalid arguments: number of operations, wires, and "
+                "parameters must all be equal");
+
+        for (size_t i = 0; i < numOperations; i++) {
+            applyOperation(ops[i], wires[i], inverse[i], params[i]);
+        }
+    }
+    /**
+     * @brief Apply multiple gates to the state-vector.
+     *
+     * @param ops Vector of gate names to be applied in order.
+     * @param wires Vector of wires on which to apply index-matched gate name.
+     * @param inverse Indicates whether gate at matched index is to be inverted.
+     * @param params Optional parameter data for index matched gates.
+     */
+    void applyOperations(const vector<string> &ops,
+                         const vector<vector<size_t>> &wires,
+                         const vector<bool> &inverse) {
+        const size_t numOperations = ops.size();
+        if (numOperations != wires.size())
+            throw std::invalid_argument(
+                "Invalid arguments: number of operations, wires, and "
+                "parameters must all be equal");
+
+        for (size_t i = 0; i < numOperations; i++) {
+            applyOperation(ops[i], wires[i], inverse[i]);
+        }
+    }
+
+    /**
+     * @brief Get indices not participating in operation.
+     *
+     * @param indicesToExclude
```

Comment:
Missing some description here?

Reply:
Yea, the docs need to follow. We have a story created for this, as the entire class docs need an update.